context.go

```diff
@@ -204,9 +204,10 @@ func (cCtx *Context) checkRequiredFlags(flags []Flag) requiredFlagsErr {
 			var flagPresent bool
 			var flagName string
 
-			for _, key := range f.Names() {
-				flagName = key
+			flagNames := f.Names()
+			flagName = flagNames[0]
```

Comment:
what happens if someone passes in a flag with no name or aliases defined ? Will the slice be nil ?

Reply:
Good catch 🦾

Reply:
It returns an empty slice (length 0), not a nil. I can make a test case for it to prove that it doesn't panic if you want, even though the code doesn't handle this kind of flag nicely (showing as a newline in the help).

What would a flag without a name or alias means to the user? Shouldn't the code have a different check and error for this case?

Reply:
if len(flagNames) == 0 ln 208 will panic:

https://go.dev/play/p/aN-gKini_p9

Reply:
@skelouse it's length 1 with an empty string. Sorry for the confusion. https://go.dev/play/p/tsnBea9RKk4 - No panic

Reply:
Both examples provided are illustrative but its better to add a simple test case for this. Just have a flag without a name and see what happens. If everything works well and good.

Reply:
Sure. Added another test case